setup.py

```diff
@@ -1,31 +1,3 @@
 from setuptools import setup
 
-from nats.aio.client import __version__
-
-EXTRAS = {
-    'nkeys': ['nkeys'],
-}
-
-setup(
-    name='nats-py',
-    version=__version__,
-    description='NATS client for Python',
-    long_description='Python client for NATS, a lightweight, high-performance cloud native messaging system',
-    classifiers=[
-        'Intended Audience :: Developers',
-        'Programming Language :: Python',
-        'Programming Language :: Python :: 3',
-        'Programming Language :: Python :: 3.7',
-        'Programming Language :: Python :: 3.8',
-        'Programming Language :: Python :: 3.9',
-        'Programming Language :: Python :: 3.10'
-        ],
-    url='https://github.com/nats-io/nats.py',
-    author='Waldemar Quevedo',
-    author_email='wally@synadia.com',
-    license='Apache 2 License',
-    packages=['nats', 'nats.aio', 'nats.protocol', 'nats.js'],
-    package_data={"nats": ["py.typed"]},
-    zip_safe=True,
-    extras_require=EXTRAS
-)
+setup()
```

Comment:
Not very clear at the moment whether GitHub supports it yet for its dependency graph: https://github.com/community/community/discussions/6456

Really like the changes, but also wonder if we keep some of the basic metadata here still for a bit like Flask is doing:

https://github.com/pallets/flask/blob/main/setup.py

So maybe reduce it to something like this?

```python
# Metadata goes in pyproject.toml. These are here for GitHub's dependency graph.
setup(
    name="nats-py",
    extras_require={
        'nkeys': ['nkeys']
    }
)
```

Reply:
Github's dependency graph support `pyproject.toml`, please see https://docs.github.com/en/code-security/supply-chain-security/understanding-your-software-supply-chain/about-the-dependency-graph#supported-package-ecosystems

However, if you prefer, it I'm not against managing the `extras_require` the way Flask is doing it.

Reply:
We should notice that Flask doesn't rely on `pyproject.toml` but rather on `setup.cfg`.

Concerning the dependency graph it seems to require a `poetry.lock` file, which is a related the poetry package manager. We don't rely on poetry here, so, if we want to keep this github feature available I'd also suggest to keep `extras_require` in the `setup.py` file.

I'll submit a new patch set to restore this point.

Thanks for the heads up.

Reply:
Done

Reply:
> Not very clear at the moment whether GitHub supports it yet for its dependency graph: [community/community#6456](https://github.com/orgs/community/discussions/6456)
> 
> Really like the changes, but also wonder if we keep some of the basic metadata here still for a bit like Flask is doing:
> 
> https://github.com/pallets/flask/blob/main/setup.py
> 
> So maybe reduce it to something like this?
> 
> ```python
> # Metadata goes in pyproject.toml. These are here for GitHub's dependency graph.
> setup(
>     name="nats-py",
>     extras_require={
>         'nkeys': ['nkeys']
>     }
> )
> ```

You should notice that currently those `extras_require` available in `setup.py` aren't listed in your current dependency graph https://github.com/nats-io/nats.py/network/dependencies.

Only those given by the pipenv artifacts are listed.

Reply:
Instead I'd suggest to use the extras mechanisms of `requirements.txt` defined by the PEP 508 [1] to manage those extras require.
`requirements.txt` is a well know mechanism where extras require can be defined.

It would lead us to:

```
nkeys [auth] >= 0.1.0
```

Instead of naming this extras `auth`, we could name it `security` by example, given the nature of the nkeys lib.
[1] https://peps.python.org/pep-0508/

Reply:
Github's dependency graph properly detect `the requirements.txt` file. Here is an example with one of my other project:
https://github.com/openuado/niet/network/dependencies